src/rgw/rgw_data_sync.cc

```diff
@@ -1464,12 +1469,13 @@ class RGWDataSyncShardCR : public RGWCoroutine {
   RGWCoroutine* sync_single_entry(const rgw_bucket_shard& src,
                                   const std::string& key,
                                   const std::string& marker,
-                                  ceph::real_time timestamp, bool retry) {
+                                  ceph::real_time timestamp, bool retry,
+				  std::optional<uint64_t> entry_gen) {
```

Comment:
sync always needs to know the generation number here, so this shouldn't be optional. it looks like we're still missing that from async notifications and the error_repo, so we should figure out how to add it there too

Reply:
Does it make sense to send it to the repo for retry, then? If the generation number is old, then it would just error out again.

Reply:
`struct rgw_bucket_sync_status` will need to track the gen it's currently syncing. if we see something older than that current gen, we can just discard it

Reply:
Done.